### DISCUSSION TITLE
What files do "ctu_connections" actually contain?

### DISCUSSION THREAD

**Discussion by sahil-time** [Ericsson/codechecker#4220] [Q&A] [answered]
Hi folks,

- I was checking the "ctu_connections" folder, and I'm not sure how particular files are chosen for a TU
- For instance, I see some files that are directly needed for the current translation unit, and some files that are indirectly needed. 
- How does CodeChecker decide this?

Thanks,

**Comment by whisperity** [comment, marked as answer]
It's not CodeChecker's decision, we only read the list of files the analyser actively looked into during an executed analysis:

https://github.com/Ericsson/codechecker/blob/d2b1197810ece5f219ead596a220b4fc49313213/analyzer/codechecker_analyzer/analysis_manager.py#L461-L482
